### ISSUE TITLE
Support Unix sockets

### ISSUE THREAD

**Issue by moll** [rlidwka/sinopia#237]
Hey,

Unix sockets are a little more versatile and easier to lock down than TCP for reverse proxying. Node usually, where it takes a port number, allows a file path to be given for listening as a Unix socket. Sinopia does something clever with the port, however, and forbids this:

```
$ sinopia --listen sinopia.sock
 warn  --- invalid address - null, we expect a port (e.g. "4873"), host:port (e.g. "localhost:4873") or full url (e.g. "http://localhost:4873/")
```

It'd be nice to be able to use Unix sockets, too.

**Comment by ppalladino** [comment]
Moll, I also put in a request to extend port configuration. https://github.com/rlidwka/sinopia/issues/236 I think we are kinda, sorta, looking for a similar thing.....

**Comment by rlidwka** [comment]
> Node usually, where it takes a port number, allows a file path to be given for listening as a Unix socket.

Node doesn't allow something like `0.0.0.0:1234` as a listening address, it requires host and port to be separate.

I tried to squeeze host and port together. Other people requested protocol as well. So we ended up with an ability to do any of [these](https://github.com/rlidwka/sinopia/blob/e8593c47cc380474f005ea20b606adcbabc235f3/lib/utils.js#L129-L134).

I think `unix:sinopia.sock` would be appropriate to add.
